src/lib/FreeC/IR/Subst.hs

```diff
@@ -159,6 +159,7 @@ instance ApplySubst IR.Expr IR.Expr where
       let (subst', args') = newRenameArgs subst args
           expr'           = applySubst subst' expr
       in  IR.Lambda srcSpan args' expr' exprType
+    applySubst' expr@(IR.Let _ _ _ _) = expr -- TODO needed ?
```

Comment:
The case for `Let` should be analogous to `Lambda`.
Let-expression can be expressed as lambda expressions as follows (if the let-binding is not recursive).

```haskell
let x = f 1 True
in x + 42
```
is the same as
```haskell
(\x -> x + 42) (f 1 True)
```

Reply:
Oh, i missed that. I will keep that in mind when implementing the other functions for let. Thanks!